code/game/objects/structures/iv_drip.dm

```diff
@@ -1,60 +1,77 @@
 /obj/structure/iv_drip
 	name = "\improper IV drip"
 	icon = 'icons/obj/structures/iv_drip.dmi'
-	anchored = 0
-	density = 0
+	anchored = FALSE
+	density = FALSE
+
 	var/mob/living/carbon/human/attached
 	var/mode = 1 // 1 is injecting, 0 is taking blood.
 	var/obj/item/chems/beaker
 	var/list/transfer_amounts = list(REM, 1, 2)
 	var/transfer_amount = 1
 
+/obj/structure/iv_drip/Initialize()
+	. = ..()
+	queue_icon_update()
+
 /obj/structure/iv_drip/verb/set_amount_per_transfer_from_this()
 	set name = "Set IV transfer amount"
 	set category = "Object"
 	set src in range(1)
 	if(!CanPhysicallyInteract(usr))
-		to_chat(usr, "<span class='notice'>You're in no condition to do that!'</span>")
+		to_chat(usr, SPAN_NOTICE("You're in no condition to do that!"))
 		return
 	var/N = input("Amount per transfer from this:","[src]") as null|anything in transfer_amounts
 	if(!CanPhysicallyInteract(usr)) // because input takes time and the situation can change
-		to_chat(usr, "<span class='notice'>You're in no condition to do that!'</span>")
+		to_chat(usr, SPAN_NOTICE("You're in no condition to do that!"))
 		return
 	if(N)
 		transfer_amount = N
 
 /obj/structure/iv_drip/on_update_icon()
-	if(attached)
-		icon_state = "hooked"
-	else
-		icon_state = ""
+	cut_overlays()
 
-	overlays.Cut()
+	var/mutable_appearance/base = mutable_appearance(icon, "nothing")
+	base.icon_state = "[!!beaker && "beaker" || "nothing"][!!attached && "_hooked" || ""]"
```

Comment:
This would be clearer as `"[beaker ? "beaker" : "nothing"][attached ? "_hooked" : ""]"` imo.